File: functions/libsodium/variants/c/4_fBlaMka_0.c
#include <stdint.h>

static inline uint64_t fBlaMka(uint64_t x, uint64_t y)
{
    return x + y + 2 * (uint64_t)((uint32_t)x * (uint32_t)y);
}


int main(){
    fBlaMka(0,0);
}

File: functions/libsodium/variants/go/4_fBlaMka_0.go
package main

func fBlaMka(x, y uint64) uint64 {
	const m uint64 = 0xFFFFFFFF
	xy := (x & m) * (y & m)
	return x + y + 2*xy
}


func main(){
    fBlaMka(0,0)
}
